File: src/main/java/com/example/javakotlindemo/JavaKotlinDemoApplication.java
package com.example.javakotlindemo;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class JavaKotlinDemoApplication {

    public static void main(String[] args) {

        SpringApplication.run(JavaKotlinDemoApplication.class, args);

        HelloService helloService = new HelloService();
        System.out.println(helloService.sayHello());

    }
}

File: src/main/java/com/example/javakotlindemo/JavaKotlinDemoApplication.kt
package com.example.javakotlindemo

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication

@SpringBootApplication
object JavaKotlinDemoApplication {
    @JvmStatic
    fun main(args: Array<String>) {
        SpringApplication.run(JavaKotlinDemoApplication::class.java, *args)
        val helloService = HelloService()
        println(helloService.sayHello())
    }
}